### DISCUSSION TITLE
Deskflow Server <> Synergy Client: About Compatibility

### DISCUSSION THREAD

**Discussion by Dentrax** [deskflow/deskflow#7926] [Q&A] [answered]
I've been considering purchasing the Synergy product, but I have some questions regarding compatibility. If I install Deskflow on the server PC and Synergy on another client PC (or vice-versa: Synergy Server <> Deskflow Client), would they work together? Are networking stuff / protocols fully compatible?

I'd asking this because:
* I'd like to continue using the open-source version of this product on a Linux system to provide feedback, while using the premium version on Windows or Mac.
* To support the company and developers

**Comment by sithlord48** [comment]
Yes they should work, 
Deskflow has an optiton to talk to either synergy or barrier (deskflow, input-leap and barrier use this) 
Synergy should being downsteam eventually gain that ability also but to be sure we better ask @nbolton .

**Reply by Dentrax** [reply]
Thanks, I just bought the Synergy 3 Ultimate license, I able to run it on the Server, and actually I don't know what should I configure on the client's deskflow configuration. I can't able to move the mouse to other display. If I check with `lsof`, Synergry listens some ports: `24800`, `24802` and `24803`. Thanks. cc @nbolton

**Reply by Dentrax** [reply]
Client running on this IP, but it says "Connection Failed", probably the app expects a specific port or verification header, etc.

![image](https://github.com/user-attachments/assets/90063fa6-e1d1-4894-bab7-8d16794ada7a)

Client logs are:

```
[2024-11-28T21:31:09] ERROR: server refused client with name "Macbook-Pro-1"
[2024-11-28T21:31:09] WARNING: failed to connect to server: server refused client with our name
```

Server logs are:

```
core       [2024-11-28T21:39:52] - NOTE    - accepted client connection
core       [2024-11-28T21:39:52] - INFO    - local languages: tr, en
core       [2024-11-28T21:39:52] - WARNING - unrecognised client name "Macbook-Pro-1", check server config
core       [2024-11-28T21:39:52] - NOTE    - disconnecting client "Macbook-Pro-1"
core       [2024-11-28T21:39:52] - NOTE    - client "Macbook-Pro-1" has disconnected
```

**Reply by Dentrax** [reply, marked as answer]
After some debugging, I able to make it work! I have changed the client name on the GUI, and set it to what included in the `/Users/furkan/Library/Preferences/Synergy/synergy.conf`. (But had to install Synergy on the client and then uninstalled. Synergy keep crashed on manual config updates)

But I hit lots of `event queue is not ready` errors: https://github.com/deskflow/deskflow/issues/4276

**Reply by samiralam** [reply]
Can you explain more on how you got this working? I have a Linux Deskflow client and a Mac Synergy 3 server, and when I try to connect one to the other I get the "unrecognized client name" error. I tried editing `synergy.conf` on the server but my changes don't persist.

**Reply by sithlord48** [reply]
the settings will be overwritten if the server/ client or gui is running so be sure to close them all before any hand editing ., its best to change things in the gui if you can for clients you can edit them by double clicking on them in the server configs ' computer tab .

**Reply by samiralam** [reply]
Progress! I was able to get the two connected by directly editing the Synergy config file. But I can't seem to add a Deskflow client via the GUI as you described. Doing it by IP or hostname in "Manually add computer" just results in the "Connection failed" error that @Dentrax screenshotted above.

Is there a way to add/rearrange Deskflow clients via the Synergy 3 GUI?

**Reply by nbolton** [reply]
> Is there a way to add/rearrange Deskflow clients via the Synergy 3 GUI?

Unfortunately, there are no plans to make Synergy 3 and Deskflow compatible. Only Synergy 1 and Deskflow are compatible.

**Reply by samiralam** [reply]
> Unfortunately, there are no plans to make Synergy 3 and Deskflow compatible. Only Synergy 1 and Deskflow are compatible.

That clears it up, thanks. Ideally I'd just use Synergy 3 on both machines, but there's currently no release available for Arch Linux. Any idea if there's a flatpak in the works?

**Reply by nbolton** [reply]
For Arch, how about [the Synergy 3 AUR](https://aur.archlinux.org/packages/synergy3-bin)? `yay synergy3-bin`

We do have plans to ship both an official Arch package and Flatpak for Synergy 3, but I can't say when.
